### PR TITLE
 SPECTUB: initialising old global values: fixing #1232

### PR DESCRIPTION
**Pull request by danieldeidda**

### CODE REVIEW

File: src/recon_buildblock/ProjMatrixByBinSPECTUB.cxx [hunk 1]
```diff
@@ -393,30 +394,31 @@ set_up(
...
 	}
 	
-	if ( wmh.do_psf ){ 		
-	  wmh.predef_col = false;
+      wmh.predef_col = false;
 	  wmh.COL.A = collimator_slope;
 	  wmh.COL.B = collimator_sigma_0;
-	}
-	else{
+      wmh.COL.do_fb = false;
```
Comment (KrisThielemans) [inline src/recon_buildblock/ProjMatrixByBinSPECTUB.cxx:400]:
```suggestion
      // no fan-beam
      wmh.COL.do_fb = false;
      // code to enable fan-beam collimator at some point... (not validated)
      // before enabling, you will have to change the parameter file to give parameters of the fan-beam
      // wmh.COL.do_fb = true;
      // wmh.COL.F     = collimator_number;
```

File: src/recon_buildblock/ProjMatrixByBinSPECTUB.cxx [hunk 2]
```diff
@@ -393,30 +394,31 @@ set_up(
...
+
+    if ( !wmh.do_psf )
  	  // code to enable fan-beam collimator at some point... (not validated)
 	  // before enabling, you will have to change the parameter file to give parameters of the fan-beam
 	  //int num = collimator_number;
-	  //if ( num ==0 ) {
-	  wmh.COL.do_fb = false;
+      //if ( num ==0 ) {
```
Comment (KrisThielemans) [inline src/recon_buildblock/ProjMatrixByBinSPECTUB.cxx:406]:
```suggestion
      //if (wmh.COL.do_fb ==0 ) {
```

**Review by KrisThielemans — CHANGES_REQUESTED** [review]
looks good. some comments on clean-up.

Did you run this through `valgrind`?

**Comment by KrisThielemans** [comment]
@samdporter please take a look and try

**Comment by samdporter** [comment]
> @samdporter please take a look and try

Looks good to me!

Tested with STIR & SIRF

**Comment by KrisThielemans** [comment]
Thanks @samdporter. 'valgrind' as well?

**Comment by samdporter** [comment]
> Thanks @samdporter. 'valgrind' as well?

Looks fine to me. Updated the valgrind output files [here](https://github.com/samdporter/SPECTissues)

**Comment by KrisThielemans** [comment]
Can either of you look at my comments please? Would like to merge this soon

**Review by KrisThielemans — APPROVED** [review]

**Pull request merged**